File: stage1/problems/problem4/starter.cpp
#include <iostream>
#include <vector>
#include <string>
#include <sstream>

void solve() {
    // --- Input reading ---
    std::string msg, words_line;
    std::getline(std::cin, msg);
    std::getline(std::cin, words_line);
    
    std::stringstream ss(words_line);
    std::string word;
    std::vector<std::string> words_to_append;
    while(ss >> word) {
        words_to_append.push_back(word);
    }

    // --- Solution ---
    // The following variables are available:
    // msg: std::string
    // words_to_append: std::vector<std::string>

    // TODO: Implement the solution
}

int main() {
    std::ios_base::sync_with_stdio(false);
    std::cin.tie(NULL);
    solve();
    return 0;
}

File: stage1/problems/problem4/starter.py
import sys

def solve():
  """
  Reads input from stdin, solves the problem, and prints the output.
  """
  try:
    # --- Input reading ---
    msg = sys.stdin.readline().strip()
    words_to_append = sys.stdin.readline().strip().split()
    
    # --- Solution ---
    # The following variables are available:
    # msg: str
    # words_to_append: list of strings
    
    # TODO: Implement the solution
    
  except (ValueError, IndexError):
    pass

if __name__ == "__main__":
    solve()
